newIDE/app/src/EventsFunctionsExtensionEditor/EventsFunctionConfigurationEditor/EventsFunctionParametersEditor.js

```diff
@@ -50,6 +50,17 @@ const styles = {
   },
 };
 
+const parseJSONArray = (json, callback) => {
```

Comment:
nitpicking: return the array rather than having a callback, it's not necessary here

Reply:
It is what I had originally done, but as it is used inside the JSX it is easier and imo cleaner this way as I only parse it once. Else I would need to reparse multiple times the json.

Reply:
In this case, maybe it would a good call for moving this into a separate component?

> Else I would need to reparse multiple times the json.

Well you're already parsing it again when adding a new line, so you can also parse it again when applying the changes made on a text field?

Reply:
> In this case, maybe it would a good call for moving this into a separate component

I am afraid this might be a bit too specific to be reused somewhere else?

> Well you're already parsing it again when adding a new option

Oh, that was a mistake, I can fix that.

Reply:
You can still move this to a component but not in a separate file:

```
type SelectorListEditorProps = {| ... |};
const SelectorListEditor = (props: SelectorListEditorProps) => {
}
```

The advantage being that you can offer a clean interface for props:
* `extraInfo: string`
* `onChangeExtraInfo: () => string`

which should be almost all that is needed? 
So the already huge component is not too much impacted by this.

Reply:
I am scared splitting the code like that might just make it harder to read...